Add new SeekBar to rotate Image

nakatalog opened the pull request:
Add new SeekBar to rotate Image
Change SeekBar to more general Use

nakatalog committed:
Add new SeekBar to rotate Image

koji-1009 commented on app/src/main/java/com/app/dr1009/colorpickers/PickerData.kt:
```diff
@@ -5,6 +5,7 @@ import android.databinding.Bindable
 
 class PickerData : BaseObservable() {
 
+    // region Title Of SeekBar
```
下記理由から、定義場所の宣言は必要ではないと考えています。

- fieldの宣言場所を明示しなければならないほど多くのfieldを保持していないこと
- 一見したとき、コメントがfieldの説明に見えてしまうため、紛らわしいこと
- fieldの名称から自明な情報であること

nakatalog replied:
承知しました。region ~ endregionは削除します。

koji-1009 commented on app/src/main/java/com/app/dr1009/colorpickers/PickerData.kt:
```diff
@@ -18,8 +19,25 @@ class PickerData : BaseObservable() {
     val back = RGBAData(aDefault = 50, rDefault = 60, gDefault = 60, bDefault = 60)
+
+    @get:Bindable
+    val rotation = RotationData(rotateDefault = 0)
```
この ``rotateDefault`` が仮引数の値であるならば、明示して宣言時にセットする必要はないと考えます。

nakatalog replied:
それもそうですね。仮引数を使うなら明示する必要がないですね。

koji-1009 commented on app/src/main/java/com/app/dr1009/colorpickers/RotationData.kt:
```diff
@@ -0,0 +1,15 @@
+import android.databinding.BaseObservable
+import android.databinding.Bindable
+
+class RotationData(rotateDefault: Int = 0) : BaseObservable() {
```
rotationはColorPickerに対して1つの属性にあたるように感じます。
そのため、新規にクラスを作成するのではなく、PickerDataの一つのfieldとして宣言する方が適当に思うのですが、いかがでしょう。

nakatalog replied:
個人的には、回転はMatrixを使ったりすると、色々なことが可能なので、回転のみでデータクラスにしておいた方がわかりやすいかと思います。

koji-1009 replied:
なるほどです。
そちらの方が面白そうですね。ありがとうございます。

koji-1009 commented on app/src/main/res/layout/seek_item.xml:
```diff
@@ -55,7 +60,7 @@
             android:layout_marginStart="8dp"
             android:layout_marginTop="7dp"
-            android:max="255"
+            android:max="@{maxSeek}"
```
Cool !

koji-1009 commented on app/src/main/java/com/app/dr1009/colorpickers/PickerData.kt:
```diff
@@ -18,8 +19,25 @@ class PickerData : BaseObservable() {
+
+    // region Max Of SeekBar Progress
+    @get:Bindable
+    val maxRGBA = 255
```
定数定義なのでsnake_caseで定義したいと思います。
masterを修正したので、申し訳ないのですがこちらもsnake_caseでの宣言をお願いいたします。

nakatalog replied:
承知しました。スネークケースで記載します。

koji-1009 reviewed (commented):
Thanks for your cooperation !

I add some comments.
Could you please check?

koji-1009 commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -82,24 +87,35 @@
                     app:seek="@={data.back.a}"
                     app:type="@{data.typeA}" />
+
+                <include
```
TextViewで色ではなく、回転を弄る箇所であることを説明した方が、UIとしていいのではと思います。

nakatalog replied:
すみません、そうですね。修正します。

nakatalog committed:
Merge branch 'master' into master

nakatahello commented:
コンフリクト解消しないと、自分のブランチにkoji-1009さんの変更が反映されないんですね。

nakatalog committed:
Delete comments of “region - endregion”
Change constant number format from CameCase to SNAME_CASE
Add TextView And StringResource

nakatalog commented:
I resolved All Conflicts. Please Check It ! ↑

koji-1009 commented:
LGTM! Thanks!
